client/signup/main.jsx

```diff
@@ -185,13 +191,13 @@ const Signup = React.createClass( {
 	componentDidMount() {
 		debug( 'Signup component mounted' );
 		SignupProgressStore.on( 'change', this.loadProgressFromStore );
-		SignupProgressStore.on( 'change', this.loadDependenciesFromStore );
+		this.unsubscribeSignupDepStoreListener = SignupDependencyStore.reduxStore.subscribe( this.loadDependenciesFromStore );
```

Comment:
cc @aduth @mtias are we ok with this approach here vs building up a redux subtree in parallel and eventually switching over with with a feature flag? I'm a bit wary of doing it this way since the this handler will be triggered on **any** global state change, not just this subtree.